lib/rucio/common/schema/generic_multi_vo.py

```diff
@@ -0,0 +1,415 @@
+# Copyright 2017-2019 CERN for the benefit of the ATLAS collaboration.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+# - Vincent Garonne <vgaronne@gmail.com>, 2017
+# - Hannes Hansen <hannes.jakob.hansen@cern.ch>, 2018
+# - Robert Illingworth <illingwo@fnal.gov>, 2018
+# - Mario Lassnig <mario.lassnig@cern.ch>, 2019
+# - Jaroslav Guenther <jaroslav.guenther@gmail.com>, 2019
+# - Andrew Lister <andrew.lister@stfc.ac.uk>, 2019
+#
+# PY3K COMPATIBLE
+
+from jsonschema import validate, ValidationError
+
+from rucio.common.exception import InvalidObject
+
+
+ACCOUNT = {"description": "Account name",
+           "type": "string",
+           "pattern": "^[a-z0-9-_]{1,30}$"}
+
+ACCOUNTS = {"description": "Array of accounts",
+            "type": "array",
+            "items": ACCOUNT,
+            "minItems": 0,
+            "maxItems": 1000}
+
+
+ACCOUNT_TYPE = {"description": "Account type",
+                "type": "string",
+                "enum": ["USER", "GROUP", "SERVICE"]}
+
+ACTIVITY = {"description": "Activity name",
+            "type": "string",
+            "enum": ["Data Brokering", "Data Consolidation", "Data rebalancing",
+                     "Debug", "Express", "Functional Test", "Group Subscriptions",
+                     "Production Input", "Production Output",
+                     "Analysis Input", "Analysis Output", "Staging",
+                     "T0 Export", "T0 Tape", "Upload/Download (Job)",
+                     "Upload/Download (User)", "User Subscriptions"]}
+
+SCOPE_LENGTH = 25
```

Comment:
I've just remembered our discussion about scope/account name lengths, and how the `@tst` would effectively reduce this from 25 to 21 characters. Were we going to change the schema to reflect this? As I recall we call `validate_schema` at the API level, so would we want the schema to have ` SCOPE_LENGTH = 21` as it checks the external rep, which will then ensure we have a 25 character string at the core/DB level? I'd say the same thing for account but it looks like that can be up to 30 characters based on line 32 despite the fact the DB column has a max of 25... so maybe that's checked somewhere else?